notify/impl.go

```diff
@@ -251,7 +251,7 @@ func (n *Email) auth(mechs string) (smtp.Auth, error) {
 			return LoginAuth(username, password), nil
 		}
 	}
-	return nil, nil
+	return nil, errors.New("Unknown auth mechanism: " + mechs)
```

Comment:
There are several error paths:

* the remote server offer none of CRAM-MD5, AUTH or LOGIN.
* the remote server offer supported mechanisms but the AlertManager configuration doesn't match (for instance, missing auth password with PLAIN).

I think the error message should be clear on the reason. Maybe [`type.MultiError`](https://github.com/prometheus/alertmanager/blob/b63b5600741b07795e50cedf8818a853bacc5f76/types/types.go#L211) can be useful here.

Reply:
Well, as both errors cannot append at the same time, can we consider to return another error in that case?

Reply:
Sure. Though you can still have PLAIN and CRAM-MD5 available but neither AuthSecret nor AuthPassword set.